test/e2e_bind_test.go

```diff
@@ -1,13 +1,34 @@
 package test
 
 import (
+	"fmt"
 	"net/http"
+	"os"
+	"os/exec"
 	"testing"
+	"time"
 )
 
-func TestE2EBind(t *testing.T) {
-	req, err := http.NewRequest(http.MethodGet, "http://localhost:10000/", nil)
+func startEnvoy(configPath string) {
+	cmd := exec.Command("envoy", "-c", configPath)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Start()
+	if err != nil {
+		panic(fmt.Sprintf("failed to start envoy: %v", err))
+	}
+	err = cmd.Wait()
+	if err != nil {
+		panic(fmt.Sprintf("failed to wait envoy: %v", err))
+	}
+}
 
+func TestBind(t *testing.T) {
+
+	go startEnvoy("../example/envoy.yaml")
+	time.Sleep(5 * time.Second)
```

Comment:
go + sleep seems not really good, maybe just remove the `go` and `time.Sleep`?

Reply:
But I need to wait for the application to fully start before proceeding with the testing, otherwise I might encounter connection rejections.